### ISSUE TITLE
Building packs fails on DB files with size column when using zipped ROMs

### ISSUE THREAD

**Issue by TORQUENDB** [frederic-mahe/Hardware-Target-Game-Database#696]
build_pack.py only uses the CRC hash when checking zipped files against the database, instead of the SHA256 it uses for normal files. However, when reading the DB files, it only splits by 4 tabs, and if a DB file has the new size column (e,g. SNES, Wonderswan), the hash_crc variable contains the CRC value, a tab, and the file size. This makes it impossible to use zipped ROMs with these DB files because the hash comparison always fails.

**Comment by frederic-mahe** [comment]
thanks @TORQUENDB ! With your nice description of the issue, it was easy to find the bug.

Commit e7559aa should solve the issue. Please confirm, so we can close this thread.

**Comment by JosVerheij** [comment]
I'm not sure if these issues are related, but with commit https://github.com/frederic-mahe/Hardware-Target-Game-Database/commit/e7559aa53799874cc201475dd045f5e37806f1fc I encounter the following error when running database `EverDrive GB SMDB.txt` with `build_pack.py` on a `.zip.` rompack.

```
Traceback (most recent call last):
  File "./build_pack.py", line 362, in <module>
    DROP_INITIAL_DIRECTORY)
  File "./build_pack.py", line 224, in parse_database
    hash_sha256, filename, _, _, hash_crc = line.strip().split("\t")[0:4]
ValueError: not enough values to unpack (expected 5, got 4)
```

At a quick glance it appears the abovementioned size column might now be required by the script.

Command used:

`python3 ./build_pack.py --input_folder [INPUT] --database EverDrive\ Pack\ SMDBs/EverDrive\ GB\ SMDB.txt --file_strategy smart --output_folder [OUTPUT] -m ../missing/gameboy.txt -x`

**Comment by frederic-mahe** [comment]
oops, I made a off-by-one mistake (shame on me!). That should be fixed in 4ee5e6efe657de4d953b82fd208aa4ba6b5aa2a0 Thanks @JosVerheij

**Comment by tjanas** [comment]
I tested the latest commit to build_pack.py with the GBC SMDB (which doesn't contain file sizes) and a zip file source, and it seems to work now.
